.github/workflows/integration.yml

```diff
@@ -0,0 +1,243 @@
+name: Test Orbit & Preview
+
+on:
+  workflow_dispatch: # Manual
+  schedule:
+  - cron: '0 2 * * *' # Nightly 2AM UTC
+
+jobs:
+  gen:
+    runs-on: ubuntu-latest
+    outputs:
+      subdomain: ${{ steps.gen.outputs.subdomain }}
+      address: ${{ steps.gen.outputs.address }}
+    steps:
+    - id: gen
+      run: |
+        UUID=$(uuidgen)
+        echo "::set-output name=subdomain::fleet-test-$UUID"
+        echo "::set-output name=address::https://fleet-test-$UUID.fleetuem.com"
+  
+  run-server:
+    runs-on: ubuntu-latest
+    needs: gen
+    steps:
+    - name: Start tunnel
+      env: 
+        CERT_PEM: ${{ secrets.CLOUDFLARE_TUNNEL_FLEETUEM_CERT_B64 }}
+      run: |
+        # Install cloudflared
+        wget https://github.com/cloudflare/cloudflared/releases/latest/download/cloudflared-linux-amd64.deb
+        sudo dpkg -i cloudflared-linux-amd64.deb
+        # Add secret
+        echo "$CERT_PEM" | base64 -d > cert.pem
+        # Start tunnel
+        cloudflared tunnel --origincert cert.pem --hostname ${{ needs.gen.outputs.subdomain }} --url http://localhost:1337 --name ${{ needs.gen.outputs.subdomain }} &
+        until [ $(cloudflared tunnel --origincert cert.pem info -o json ${{ needs.gen.outputs.subdomain }} | jq '.conns[0].conns[0].is_pending_reconnect') = false ]; do
+          echo "Awaiting tunnel ready..."
+          sleep 5
+        done
+
+    # Download fleet and fleetctl binaries from last successful build on main
+    - name: Download binaries
+      uses: dawidd6/action-download-artifact@09385b76de790122f4da9c82b17bccf858b9557c #v2.16.0
+      with:
+        workflow: build-binaries.yaml
+        branch: main
+        name: build
+        path: build
+        check_artifacts: true
+
+    - name: Run Fleet server
+      timeout-minutes: 15
+      env:
+        # Use instance identifier to allow for duplicate UUIDs
+        FLEET_OSQUERY_HOST_IDENTIFIER: instance
+      run: |
+        chmod +x ./build/fleetctl
+        ./build/fleetctl preview --no-hosts
+        ./build/fleetctl config set --address ${{ needs.gen.outputs.address }}
+        ./build/fleetctl get enroll-secret
+        docker compose -f ~/.fleet/preview/docker-compose.yml logs --follow fleet01 fleet02 &
+        # Wait for all of the hosts to be enrolled
+        EXPECTED=12
```

Comment:
Am a bit lost, how can it be `EXPECTED=12` if `./build/fleetctl` is being run with `--no-hosts`?

Reply:
Ah!, I haven't gotten to the bottom jobs, they run in parallel.

Reply:
It's waiting for the hosts to enroll in the other jobs (on separate machines)